### PR TITLE
Future proofs Easter calculations

### PR DESCRIPTION
**Pull request by tortellinitony**
Credit to @RemieRichards where due.

### CODE REVIEW

**Comment by JJRcop** [comment]
If you implement this change, @MrPerson doesn't have a chance at visiting the future...

**Comment by tortellinitony** [comment]
If someone cared enough about this in the future, wouldn't we have already witnessed them travel to our time? Or maybe that is just evidence that this will be merged.

File: code/__HELPERS/dates.dm [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//Curse you calenders...
+/proc/IsLeapYear(y)
+	return ((y) % 4 == 0 && ((y) % 100 != 0 || (y) % 400 == 0))
+
+//Y, eg: 2017, 2018, 2019, in num form (not string)
+//etc. Between 1583 and 4099
+//Adapted from a free algorithm written in BASIC (https://www.assa.org.au/edm#Computer)
+/proc/EasterDate(y)
```
Comment (duncathan) [inline code/__HELPERS/dates.dm]: make these procs on the Easter datum instead of global

**Review by duncathan — DISMISSED** [review]

File: code/__HELPERS/dates.dm (hunk 1 above)
Comment (tortellinitony) [reply code/__HELPERS/dates.dm]: other holidays might use it

Comment (ACCount12) [reply code/__HELPERS/dates.dm]: What other holiday might use it?

Comment (RemieRichards) [reply code/__HELPERS/dates.dm]: My original PR contained a few other holidays, there are atleast 2-3 that are defined as "whenever easter occurs + X" where X is some offset in days, weeks, etc.

**Comment by tortellinitony** [comment]
I tested it with 5 randomly picked years between 2000 and 3000 and they all worked compared to the online tool I used.

**Pull request merged**